### PR TITLE
RD-2078 Added underlying logic to DeployOnModal

### PR DESCRIPTION
**Pull request by kubama**

### CODE REVIEW

**Review by qooban — APPROVED** [review]
Approving to make it possible to merge it for tonight build.

We'll do proper review later on this week.

File: app/utils/Manager.js [hunk 1]
```diff
@@ -78,9 +78,17 @@ export default class Manager extends Internal {
...
 
             if (totalSize > cumulativeSize) {
-                return this.doGetFull(url, params, parseResponse, fullData, cumulativeSize);
+                return this.doFetchFull(fetcher, params, fullData, cumulativeSize);
             }
             return fullData;
         });
     }
+
+    doPostFull(url, params, requestBody) {
```
Comment (qooban) [inline app/utils/Manager.js:87]: Good idea to add it.
I'll reuse it in https://github.com/cloudify-cosmo/cloudify-stage/pull/1344 .

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx [hunk 2]
```diff
@@ -1,19 +1,84 @@
...
 }
 
-const DeployOnModal: FunctionComponent<DeployOnModalProps> = ({ toolbox, onHide }) => {
+const headerT = (suffix: string) => Stage.i18n.t(`${i18nPrefix}.header.${suffix}`);
+
+const DeployOnModal: FunctionComponent<DeployOnModalProps> = ({ filterRules, toolbox, onHide }) => {
+    function fetchEnvironments() {
+        return toolbox
+            .getManager()
+            .doPostFull('/searches/deployments', { _include: 'id' }, { filter_rules: filterRules })
```
Comment (qooban) [inline widgets/common/src/deploymentsView/header/DeployOnModal.tsx:20]: Within https://github.com/cloudify-cosmo/cloudify-stage/pull/1344 I'm adding `SearchActions` (https://github.com/cloudify-cosmo/cloudify-stage/pull/1344/files#diff-ce42a298b5a51d2464e1b518ddbb938bcba654b18ea41c6e49166aa100603a44). Can be reused here possibly.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx [hunk 3]
```diff
@@ -1,19 +1,84 @@
...
+            .then((response: DeploymentsResponse) => response.items.map(item => item.id));
+    }
+
+    function createDeploymentGroup(environements: string[], deploymentParameters: BlueprintDeployParams) {
+        return toolbox
+            .getManager()
+            .doPut(
+                `/deployment-groups/BATCH_ACTION_${moment().format('YYYY-MM-DD-HH-mm-ss-SS')}`,
```
Comment (qooban) [inline widgets/common/src/deploymentsView/header/DeployOnModal.tsx:28]: That we can also unify.

1. fetch function - see `DeploymentGroupsActions.ts` (https://github.com/cloudify-cosmo/cloudify-stage/pull/1344/files#diff-702e708cbc06623c9b840a3dc9b1565d0959cc8a9dd7f11da6ddda7f79f09b48) from https://github.com/cloudify-cosmo/cloudify-stage/pull/1344 .
2. group name function - we should probably have common function for getting group name. I can do that in my PR.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx [hunk 4]
```diff
@@ -1,19 +1,84 @@
...
+
+    function startInstallWorkflow(deploymentGroupId: string) {
+        return toolbox.getManager().doPost(
+            '/execution-groups',
+            {},
+            {
+                deployment_group_id: deploymentGroupId,
+                workflow_id: 'install'
+            }
+        );
```
Comment (qooban) [inline widgets/common/src/deploymentsView/header/DeployOnModal.tsx:52]: That we can also unify.

See `widgets/common/src/ExecutionGroupsActions.ts` which is already available in `master` branch.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx [hunk 5]
```diff
@@ -1,19 +1,84 @@
...
+    }
+
+    function finalize() {
+        toolbox.getEventBus().trigger('deployments:refresh');
```
Comment (qooban) [inline widgets/common/src/deploymentsView/header/DeployOnModal.tsx:56]: Probably we can also trigger `executions:refresh`.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx [hunk 6]
```diff
@@ -1,19 +1,84 @@
...
+
+const DeployOnModal: FunctionComponent<DeployOnModalProps> = ({ filterRules, toolbox, onHide }) => {
+    function fetchEnvironments() {
+        return toolbox
+            .getManager()
+            .doPostFull('/searches/deployments', { _include: 'id' }, { filter_rules: filterRules })
+            .then((response: DeploymentsResponse) => response.items.map(item => item.id));
+    }
+
+    function createDeploymentGroup(environements: string[], deploymentParameters: BlueprintDeployParams) {
```
Comment (qooban) [inline widgets/common/src/deploymentsView/header/DeployOnModal.tsx:24]: Minor thing, typo: `environements` -> `environments`

**Review by qooban — COMMENTED** [review]
As promised I did post-merge review 🙂

Added some comments, mostly related to code reuse/extraction.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx (hunk 4 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/DeployOnModal.tsx:52]: Right, will be fixed in next PR.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx (hunk 5 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/DeployOnModal.tsx:56]: Will be added.

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx (hunk 6 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/DeployOnModal.tsx:24]: Will be fixed

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx (hunk 2 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/DeployOnModal.tsx:20]: I'll address it once #1344 is merged

File: widgets/common/src/deploymentsView/header/DeployOnModal.tsx (hunk 3 above)
Comment (kubama) [reply widgets/common/src/deploymentsView/header/DeployOnModal.tsx:28]: > 1. fetch function - see `DeploymentGroupsActions.ts`

I'll address it once #1344 is merged

**Pull request merged**